### ISSUE TITLE
Proposing some changes

### ISSUE THREAD

**Issue by tunnckoCore** [Zimtente/ens-collections#9]
Heya there. I am 5848.eth!
I've been doing some work on that side before few weeks.
It was in https://github.com/tunnckoCore/ensmarkets, and just made it public.
Initially the work started at https://github.com/tunnckoCore/ensclubs-utils on April 28th, and quickly seen it's a mess lol, so I started it from scratch and from a CLI point of view, and started thinking what will be a good and easy to use format for the things i need.

I'm in open source for a decade, I like building things and know what newcomers can mistake, as seen in the issues and the PRs here. 
I jumped out of the bed at that night when 10kClubs started, and I started doing stuff... it's just my nature. I can make PRs with some updates and a CLI.

So, in above repo there's also basic CLI, which I will update, but most importantly the `add` command accepts any input file with comma/space/newline-separated list of names, and generates these files on `categories/` directory which looks like this

![image](https://user-images.githubusercontent.com/5038030/172372257-d932dc21-2fba-47a8-842d-23b465732ed7.png)

### Why?
It hosts both the information and the id for given label in a convenient way for us to base things on top of it. For example, I'm using this `categories` directory as source for getting the ids and then make snapshot files in `snapshots` directory. Plus, it's easy for not experienced "users" to add categories. We can even make an app and form that they can use to submit proper PRs, which form will not be much different than the BulkSearch on ENSVison, except with name and description. I think you got good at "slugs" on ensvision and we should not allow them to add slug. 

### The snapshots

You probably don't need that, but just an example, why i'm using `{ info: Object, data: Object }` format. `ens-collections` can be auto-generated from `categories` too. Thus, avoiding user mistakes, they just know how to put a list on Bulk Search, not familiar with all the git and GitHub stuff.

It actually creates 3 files for each club in the snapshots directory out of that single source of truth without possible user errors in the middle.

![image](https://user-images.githubusercontent.com/5038030/172372743-04bea49f-5715-408e-9fe2-e538cc4b0a1d.png)

### `0x1k.json` 

Expands from above format, and makes an object for each label. For times when you want to request by label.

![image](https://user-images.githubusercontent.com/5038030/172373222-e991d8e5-79f1-4015-8800-cbe42b26e21f.png)

### `0x1k-holders.json`

On this one, `data` is "per holder", meaning the key is the owner, and the value is all owner's labels.

![image](https://user-images.githubusercontent.com/5038030/172373838-4c8a5001-8556-44a9-aca2-091e59b7ade0.png)

### `0x1k-ranked.json`

Is where on `data` you get a list of number of labels that an owner holds. A la "holders" for ERC20 token.

![image](https://user-images.githubusercontent.com/5038030/172374136-b00e9c53-2e3b-487c-9d61-4c762da1185b.png)

**Comment by Zimtente** [comment]
Hi there,
thanks for the effort already with the other pull requests. Ive added you to the repo 👍 

I totally agree that we need to mitigate errors for "not experienced users".

I'm ok with converting the .csv format to .json. your format looks good. Not sure if it's important to have **owner** in the mix. 

@paste what do you think?

**Comment by tunnckoCore** [comment]
Well, yea, definitely. Owner and info.owners are not needed, it's just part of the snapshotting, it will be just

```
{
  info: {
    name: string, 
    description: string, 
    supply: number, (auto calculated)
    website: string, 
    ...
  }, 
  data: {
    label: tokenId
  }
```

**Comment by Zimtente** [comment]
ah, that's what you meant with snapshotting. ok yeah makes total sense. Also that the `logo` should as you said preferably be a URL.  So with your set up you would also go and remove the [ens-collections.json](https://github.com/Zimtente/ens-collections/blob/main/ens-collections.json) file?

**Comment by Zimtente** [comment]
Thats true - but some collections might have some websites/tools we might also want to collect. So might be handy to have `links` as an array as in the second example you gave.

**Comment by tunnckoCore** [comment]
That what I came from, we don't know the future upon us, so definitely better to have it right from here with arrays. [I take the idea for that by looking at the L2Beat awesome site design, layout, and source](https://github.com/l2beat/l2beat/blob/1f5c53ac9a99ee059c09c80bba1711cfe68f6076/packages/config/src/projects/arbitrum.ts#L36-L55)

**Comment by Zimtente** [comment]
Seems to look good!

Btw @Octexor made a site that would make it easy for the non-tech savvy to generate the name lists. We could add it to the description - maybe? 

Check it out here:
https://ensname2id.10ktools.eth.limo/

**Comment by Zimtente** [comment]
hey, @CryptoGatsu will help with the pull requests and the community part and help people to get it right with the repo. How should we proceed with your proposed changes? @tunnckoCore

**Comment by tunnckoCore** [comment]
heya.

I think this is good.

```ts
type Project = {
  info: {
    name: string, 
    description: string, 
    supply: number, // auto-calculated
    links: string[], // websites, and etc
    community: string[] // just social media
    logo: string, // url preferably; if not given we generate one from the symbol/slug
    symbol: string, // or `slug`, like 1HXC
  }, 
  data: {
    label: tokenId
  }
}
```

**Comment by szanuje** [comment]
@tunnckoCore @Zimtente , parsing csv file is faster compared to JSON, and some collections are pretty big. Owners of the domains are dynamic, while name and tokenId will always be the same, so I wouldn't advise these changes.

**Comment by tunnckoCore** [comment]
The owners is my thing, not including it. I'm talking about the general structure and `label: tokenId`

> parsing csv file is faster compared to JSON

Not exactly:
- there's no builtin parser in the language
- reading and parsing json file is easy, or even just with dynamic `import()`-ing it
- json is used in APIs, so you can directly request them (even if it's big, you can graphql it)
